tutorials/appengine-pusher/index.md

```diff
@@ -0,0 +1,452 @@
+            updateOnlineCount();
+        });
+
+```
+
+### Receive events
+
+Clients can receive events triggered over a channel by
+[binding](https://pusher.com/docs/client_api_guide/client_events#bind-events) to the channel
+using the event name and attaching an event handler that is triggered on receiving the event.
+
+The following example demonstrates attaching an event callback to a user-triggered event.
+In the case of the chat application, the event callback is used to update the messages
+displayed by the chat application.
+
+[embedmd]:# (java/src/main/webapp/WEB-INF/view/chat.jsp /\/\/ bind to successful subscription/ /}\);/)
+```jsp
+// bind to successful subscription
+        channel.bind('pusher:subscription_succeeded', function (members) {
+            // receive list of members on this channel
+            var whosonline_html = '';
+            members.each(function (member) {
+                whosonline_html += '<li class="chat_widget_member" id="chat_widget_member_'
+                    +
+                    member.id + '">' + member.info.displayName + '</li>';
+            });
+```
+
+
+### Trigger server-side endpoint to send messages
+
+Now you can use a server-side endpoint as described earlier to trigger an event on a chat message.
+The client can be excluded from receiving the broadcast message by providing the socket ID.
+
+[embedmd]:# (java/src/main/webapp/WEB-INF/view/chat.jsp /\/\/ track socket_id/ /false;\n.*}\);/)
+```jsp
+// track socket_id to exclude recipient in subscription
+        socket_id = pusher.connection.socket_id;
+
+        // submit the message to /chat
+        $('#chat_widget_form').submit(function () {
+            var chat_widget_input = $('#chat_widget_input'),
+                chat_widget_button = $('#chat_widget_button'),
+                message = chat_widget_input.val(); //get the value from the text input
+            var data = JSON.stringify({
+                message: message,
+                channel_id: channel_name,
+                socket_id: socket_id
+            });
+            // trigger a server-side endpoint to send the message via Pusher
+            $.post('/message', data,
+                function (msg) {
+                    chat_widget_button.show(); //show the chat button
+                    if (msg.status == "SUCCESS") {
+                        chat_widget_input.val('');
+                        handleMessage(msg); //display the message
+                    } else {
+                        alert("Error sending chat message : " + msg.status);
+                    }
+                }, "json");
+
+            return false;
+        });
+```
+
+## Disconnecting from Pusher
+
+Pusher automatically closes connections when a user navigates to another web page or closes their
+web browser.
+If you need to close a client connection manually,
+refer to [this page](https://pusher.com/docs/client_api_guide/client_connect#disconnecting).
```

Comment:
Change to "refer to Pusher's [Disconnecting from Pusher](https://pusher.com/docs/client_api_guide/client_connect#disconnecting) documentation"